3rdparty/mbedtls/CMakeLists.txt

```diff
@@ -73,21 +73,45 @@ ExternalProject_Add(mbedtls-wrap
 ExternalProject_Get_property(mbedtls-wrap SOURCE_DIR)
 ExternalProject_Get_property(mbedtls-wrap BINARY_DIR)
 
+# If we are going to rename the symbols, then we are also going to change the
+# library name, so an user can use their own mbedtls library.
+if (OE_PREFIX_MBEDTLS_SYMBOLS)
+  set(MBEDCRYPTO_LIB_NAME liboembedcrypto.a)
+  set(MBEDX509_LIB_NAME liboembedx509.a)
+  set(MBEDTLS_LIB_NAME liboembedtls.a)
+else()
+  set(MBEDCRYPTO_LIB_NAME libmbedcrypto.a)
+  set(MBEDX509_LIB_NAME libmbedx509.a)
+  set(MBEDTLS_LIB_NAME libmbedtls.a)
+endif()
+
+# Prefix the mbedtls symbols if desired.
+if (OE_PREFIX_MBEDTLS_SYMBOLS)
```

Comment:
Instead of building `oembed*.lib` and `oecrypto.lib` as substitutes to the regular version, why not have a flag that creates objcopies with the prefixes as an additional set of libs? 

- That way it doesn't disrupt the tests or sample builds, the unprefixed versions always get to run

- It opens up the possibility of us distributing these additional libraries as part of the SDK package (or as an add-on package) instead of requiring users rebuild the SDK themselves. 
   - The primary complication to this is doubling the number of pkgconfig files published just to accommodate the choice of prefixes for oecrypto/mbed* libs.